native/wasmex/src/instance.rs

```diff
@@ -1,27 +1,39 @@
-use rustler::dynamic::TermType;
-use rustler::env::{OwnedEnv, SavedTerm};
-use rustler::resource::ResourceArc;
-use rustler::types::binary::Binary;
-use rustler::types::tuple::make_tuple;
-use rustler::{Encoder, Env, Error, Term};
+use rustler::{
+    dynamic::TermType,
+    env::{OwnedEnv, SavedTerm},
+    resource::ResourceArc,
+    types::binary::Binary,
+    types::tuple::make_tuple,
+    {Encoder, Env, Error, MapIterator, Term},
+};
 use std::sync::Mutex;
 use std::thread;
 use wasmer_runtime::{self as runtime, imports};
-use wasmer_runtime_core::instance::DynFunc;
-use wasmer_runtime_core::types::Type;
+use wasmer_runtime_core::{import::Namespace, types::Type};
 
-use crate::printable_term_type::PrintableTermType;
-use crate::{atoms, functions};
+use crate::{atoms, functions, namespace, printable_term_type::PrintableTermType};
 
 pub struct InstanceResource {
     pub instance: Mutex<runtime::Instance>,
 }
 
+// creates a new instance from the given WASM bytes
+// expects the following elixir params
+//
+// * bytes (binary): the bytes of the WASM module
+// * imports (map): a map defining eventual instance imports, may be empty if there are none.
+//   structure: %{namespace_name: %{import_name: {TODO: signature}}}
```

Comment:
We should probably replace this TODO. I'll add this to my todo list.

Reply:
ohh, good catch 🌻 thanks!